kedro-datasets/setup.py

```diff
@@ -73,6 +73,9 @@ def _collect_requirements(requires):
     "spark.SparkJDBCDataSet": [SPARK, HDFS, S3FS],
     "spark.DeltaTableDataSet": [SPARK, HDFS, S3FS, "delta-spark~=1.0"],
 }
+snowpark_require = {
+    "snowflake.SnowParkDataSet": ["snowflake-snowpark-python~=1.0.0", "pyarrow>=8.0, <9.0"]
```

Comment:
Does it only work with `8.0`? This is equivalent to `pyarrow==8.0.0` since the next version after `8.0.0` is `9.0.0`

Reply:
I tested snowpark with pyarrow 10.0.0 and it is incompatible, the upper bound is actually more strict: "_please install a version that adheres to: 'pyarrow<8.1.0,>=8.0.0_"

Reply:
How about 9.0.0?

You are right technically but this is the pyarrow release ![Screenshot_20221124-012010.png](https://user-images.githubusercontent.com/18221871/203673270-a8889b30-c6b7-44fa-8712-22373cfc4879.png)

Reply:
Version 9.0.0 doesn't work, so I think we can be more specific and adjust to `pyarrow==8.0.0`

Reply:
Cool, that's fine, thanks for checking that🙏. Do you have an idea why is it not working? I think it would be nice to leave a comment there if we know certain API is not compatible.

Reply:
Sure! When creating the snowpark session it shows a warning if pyarrow version is different than 8.0.0 (**"please install a version that adheres to: 'pyarrow<8.1.0,>=8.0.0'**), and there are potential crashes when interacting with the dataframe API